ceph/ceph/templates/bin/_start_mon.sh.tpl

```diff
@@ -67,17 +67,22 @@ if [ ! -e "$MON_DATA_DIR/keyring" ]; then
     ceph-authtool $MON_KEYRING --import-keyring $keyring
   done
 
-  # Prepare the monitor daemon's directory with the map and keyring
+  # Prepare the monitor store
   ceph-mon --setuser ceph --setgroup ceph --cluster ${CLUSTER} --mkfs -i ${MON_NAME} --inject-monmap $MONMAP --keyring $MON_KEYRING --mon-data "$MON_DATA_DIR"
-else
-  log "Trying to get the most recent monmap..."
-  # Ignore when we timeout, in most cases that means the cluster has no quorum or
-  # no mons are up and running yet
-  timeout 5 ceph ${CLI_OPTS} mon getmap -o $MONMAP || true
-  ceph-mon --setuser ceph --setgroup ceph --cluster ${CLUSTER} -i ${MON_NAME} --inject-monmap $MONMAP --keyring $MON_KEYRING --mon-data "$MON_DATA_DIR"
-  timeout 7 ceph ${CLI_OPTS} mon add "${MON_NAME}" "${MON_IP}:6789" || true
 fi
 
+log "Trying to get the most recent monmap..."
+MON_IP_LIST=$(kubectl get pods --namespace=${NAMESPACE} ${KUBECTL_PARAM} -o template --template="{{`{{range .items}}`}}{{`{{if .status.podIP}}`}} {{`{{.status.podIP}}`}} {{`{{end}}`}} {{`{{end}}`}}")
+
+#If there's a quorum, we get the latest monmap
+# from one the existing Ceph monitors and
+# add ourselves to the cluster
+for mon in $MON_IP_LIST; do
+  ceph -m $mon --connect-timeout 10 ${CLI_OPTS} mon getmap -o $MONMAP || continue
```

Comment:
does it overwrite the existing monmap (even upon failure)?

Reply:
It doesn't on failure. It does when it succeed.

Reply:
what if there is just one mon? ceph -m $mon always fails to connect

Reply:
I _think_ that's fine It will fail and start with whatever it already has in $MONMAP but I'll have to double check next week